### ISSUE TITLE
monitor_keyring is undefined

### ISSUE THREAD

**Issue by Bruceforce** [ceph/ceph-ansible#3467]
**Bug Report**

What happened: monitor_keyring is undefined while executing ceph-config role. (see attached ansible.log)

What you expected to happen: monitor_keyring should be generated.

How to reproduce it (minimal and precise):

1. `git checkout stable-3.2`
1. use the the group_vars/all config provided below
1. `cp vagrant_variables.yml.sample vagrant_variables.yml`
1. `cp site.yml.sample site.yml`
1. `vagrant plugin install vagrant-vbguest`
1. install requirements `pip install -r requirements.txt`
1. start vagrant `vagrant up`

Share your group_vars files, inventory
group_vars/all
```
ceph_origin: repository
ceph_repository: community
ceph_stable_release: mimic
public_network: "192.168.42.0/24"
cluster_network: "192.168.43.0/24"
monitor_interface: eth1
devices:
  - '/dev/sdb'
  - '/dev/sdc'
osd_scenario: lvm
```

inventory is managed by vagrant.

**Environment**:
* OS (e.g. from /etc/os-release): Ubuntu 16.04 from Vagrant box ceph/ubuntu-xenial
* Kernel (e.g. `uname -a`): Linux osd0 4.4.0-22-generic #40-Ubuntu SMP Thu May 12 22:03:46 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
* Docker version if applicable (e.g. `docker version`):
* Ansible version (e.g. `ansible-playbook --version`): 2.6.11
* ceph-ansible version (e.g. `git head or tag or stable branch`): stable-3.2
* Ceph version (e.g. `ceph -v`): ceph version 13.2.2 (02899bfda814146b021136e9d8e80eba494e1126) mimic (stable)

I don't know if this is related to #2166 but setting monitor_secret manually in the group_vars/all (like monitor_secret: "AQBQn8tUsBHSIBAACVlivUvyNkn2EZREERyZRw==") fixes the problem, but I thought the secret should be generated automatically, or am I wrong?

[ansible.log](https://github.com/ceph/ceph-ansible/files/2709073/ansible.log)

**Comment by woof** [comment]
same issue

**Comment by rishabh-d-dave** [comment]
> How to reproduce it (minimal and precise):
>
>    1. checkout stable-3.2
>    2. use the the group_vars/all config provided below
 >   3. mv vagrant_variables.yml.sample vagrant_variables.yml
 >   4. install requirements pip install -r requirements.txt
 >   5. start vagrant vagrant up

I went through these steps, and `vagrant up` completed running successfully, for me. I've attached the file container `vagrant up`'s output. I used `site.yml.sample` for deploying Ceph.

@Bruceforce @woof 
Are you using a custom `site.yml`?

**Comment by Bruceforce** [comment]
@rishabh-d-dave 
I edited my initial post and included some steps I forgot previously. Now it should be clear that I'm using the sample file and just copied it.

However it seems that you are using libvirt as provider. I'm using virtualbox. I can't get ceph/ubuntu-xenial to work with libvirt because of `Waiting for domain to get an IP Address` forever. So I tried with generic/1804 box and now I don't get the `Waiting for domain to get an IP Address` issue. But my initial error is the same for libvirt and Vagrant with ceph/ubuntu-xenial AND generic/1804

See the two vagrant logs (one with virtualbox ceph/ubuntu-xenial and one with libvirt generic/1804) this time without debug logging enabled, because the issue is the same as in the initial post. 

[vagrant_libvirt.log](https://github.com/ceph/ceph-ansible/files/2718321/vagrant_libvirt.log)
[vagrant_vbox.log](https://github.com/ceph/ceph-ansible/files/2718322/vagrant_vbox.log)
